src/ShallowTraversal.js

```diff
@@ -110,7 +110,7 @@ export function buildPredicate(selector) {
           return node => nodeHasId(node, selector.slice(1));
 
         case SELECTOR.PROP_TYPE: {
-          const propKey = selector.split(/\[([a-zA-Z-]*?)(=|])/)[1];
+          const propKey = selector.split(/\[([a-zA-Z-0-9-]*?)(=|])/)[1];
```

Comment:
Something like this?
expect(wrapper.find('span[1-foo=1]')).to.have.length(0);

We can't render tags with attributes that start with numbers. <span 1-foo={1} /> will throw a parse error. Is a simple test like that sufficient or are you expecting ShallowTraversal to throw an error?